external-effects/super-blur/SuperBlur.h

```diff
@@ -0,0 +1,130 @@
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+void * factory();
+
+#ifdef __cplusplus
+}
+#endif
+
+
+namespace openshot
+{
+
+	/**
+	 * @brief This class adjusts the blur of an image, and can be animated
+	 * with openshot::Keyframe curves over time.
+	 *
+	 * Adjusting the blur of an image over time can create many different powerful effects. To achieve a
+	 * box blur effect, use identical horizontal and vertical blur values. To achieve a Gaussian blur,
+	 * use 3 iterations, a sigma of 3.0, and a radius between 3 and X (depending on how much blur you want).
+	 */
+	class SuperBlur : public EffectBase
+	{
+	private:
+		/// Init effect settings
+		void init_effect_details();
+
+		/// Internal blur methods (inspired and credited to http://blog.ivank.net/fastest-gaussian-blur.html)
+		int* initBoxes(float sigma, int n);
+		void boxBlurH(unsigned char *scl, unsigned char *tcl, int w, int h, int r);
+		void boxBlurT(unsigned char *scl, unsigned char *tcl, int w, int h, int r);
+
+
+	public:
+		Keyframe horizontal_radius;	///< Horizontal blur radius keyframe. The size of the horizontal blur operation in pixels.
+		Keyframe vertical_radius;	///< Vertical blur radius keyframe. The size of the vertical blur operation in pixels.
+		Keyframe sigma;				///< Sigma keyframe. The amount of spread in the blur operation. Should be larger than radius.
+		Keyframe iterations;		///< Iterations keyframe. The # of blur iterations per pixel. 3 iterations = Gaussian.
+
+		/// Blank constructor, useful when using Json to load the effect properties
+        SuperBlur();
+
+		/// Default constructor, which takes 1 curve. The curve adjusts the blur radius
+		/// of a frame's image.
+		///
+		/// @param new_horizontal_radius The curve to adjust the horizontal blur radius (between 0 and 100, rounded to int)
+		/// @param new_vertical_radius The curve to adjust the vertical blur radius (between 0 and 100, rounded to int)
+		/// @param new_sigma The curve to adjust the sigma amount (the size of the blur brush (between 0 and 100), float values accepted)
+		/// @param new_iterations The curve to adjust the # of iterations (between 1 and 100)
+        SuperBlur(Keyframe new_horizontal_radius, Keyframe new_vertical_radius, Keyframe new_sigma, Keyframe new_iterations);
+
+		/// @brief This method is required for all derived classes of EffectBase, and returns a
+		/// modified openshot::Frame object
+		///
+		/// The frame object is passed into this method, and a frame_number is passed in which
+		/// tells the effect which settings to use from its keyframes (starting at 1).
+		///
+		/// @returns The modified openshot::Frame object
+		/// @param frame The frame object that needs the effect applied to it
+		/// @param frame_number The frame number (starting at 1) of the effect on the timeline.
+		std::shared_ptr<Frame> GetFrame(std::shared_ptr<Frame> frame, int64_t frame_number);
+
+		/// Get and Set JSON methods
+		string Json(); ///< Generate JSON string of this object
```

Comment:
This should be fully specified as `std::string` (and throughout).

Reply:
Done